src/iris/bin/sender.py

```diff
@@ -535,6 +527,19 @@ def aggregate(now):
                 logger.info('[-] purged %s from messages %s remaining', active_message_ids, len(messages))
             del queues[key]
             sent[key] = now
+    # remove inactive message from the queue
+    connection = db.engine.raw_connection()
+    cursor = connection.cursor()
+    cursor.execute('SELECT `id` FROM `message` WHERE active=1')
+    active_message_ids = {r[0] for r in cursor}
+    cursor.close()
+    connection.close()
+    inactive = 0
+    for message_id in messages:
+        if message_id not in active_message_ids:
+            del messages[message_id]
```

Comment:
probably should do `messages.pop(message_id, None)` to avoid a KeyError in case it isn't present in that dict